source/encryption.rst

```diff
@@ -0,0 +1,145 @@
+
+      $ kubectl exec -it vault-0 -- /bin/sh
+      $ vault login s.VgQvaXl8xGFO1RUxAPbPbsfN
+      $ vault secrets enable --version=1 -path=pxc-secret kv
+
+   .. note:: You can also enable audit, which is not mandatory, but useful:
+
+      .. code:: bash
+
+         $ vault audit enable file file_path=/vault/vault-audit.log
+
+2. To enable Vault secret within Kubernetes, create and apply the YAML file,
+   as described further.
+
+   2.1. To access the Vault server via HTTP, follow the next YAML file example:
+
+      .. code:: text
+
+         apiVersion: v1
+         kind: Secret
+         metadata:
+           name: some-name-vault
+         type: Opaque
+         stringData:
+           keyring_vault.conf: |-
+              token = s.VgQvaXl8xGFO1RUxAPbPbsfN
+              vault_url = vault-service.vault-service.svc.cluster.local
+              secret_mount_point = pxc-secret
+
+      .. note:: the ``name`` key in the above file should be equal to the
+         ``spec.vaultSecretName`` key from the ``deploy/cr.yaml`` configuration
+         file.
+
+   2.2. To turn on TLS and access the Vault server via HTTPS, you should do two more things:
+      
+      * add one more item to the secret: the contents of the ``ca.cert`` file
+        with your certificate,
+      * store the path to this file in the ``vault_ca`` key.
+
+      .. code:: text
+
+         apiVersion: v1
+         kind: Secret
+         metadata:
+           name: some-name-vault
+         type: Opaque
+         stringData:
+           keyring_vault.conf: |-
+             token = = s.VgQvaXl8xGFO1RUxAPbPbsfN
+             vault_url = vault-service.vault-service.svc.cluster.local
+             secret_mount_point = pxc-secret
+             vault_ca = /etc/mysql/vault-keyring-secret/ca.cert
+           ca.cert: |-
+             ..........
+
+      .. note:: For techincal reasons the ``vault_ca`` key should either exist
+         or not exist in the YAML file; commented option like
+         ``#vault_ca = ...`` is not acceptable.
+
+More details on how to install and configure Vault can be found `in the official documentation <https://learn.hashicorp.com/vault?track=getting-started-k8s#getting-started-k8s>`_.
+
+.. _vault-encryption:
+
+Using the encryption
+--------------------
+
+If using Percona XtraDB Cluster 5.7, you should turn encryption on explicitly
+when you create a table or a tablespace. This can be done by adding the
+``ENCRYPTION='Y'`` part to your SQL statement, like in the following example:
+
+   .. code:: sql
+
+      CREATE TABLE t1 (c1 INT, PRIMARY KEY pk(c1)) ENCRYPTION='Y';
+      CREATE TABLESPACE foo ADD DATAFILE 'foo.ibd' ENCRYPTION='Y';
+
+.. note:: See more details on encryption in Percona XtraDB Cluster 5.7 `here <https://www.percona.com/doc/percona-xtradb-cluster/5.7/management/data_at_rest_encryption.html>`_.
+
+If using Percona XtraDB Cluster 8.0, the encryption is turned on by default.
```

Comment:
we can remove whole "Using the encryption" section, as soon as @tduzan provide list of options for my.cnf